internal/containers/graph/builder.go

```diff
@@ -0,0 +1,311 @@
+func (b *Builder) processMatchesContainer(cgroupContent string, container ContainerInfo) bool {
+	// Parse cgroup lines looking for container patterns
+	lines := strings.Split(strings.TrimSpace(cgroupContent), "\n")
+
+	for _, line := range lines {
+		// cgroup format: hierarchy-id:controller-list:cgroup-path
+		parts := strings.SplitN(line, ":", 3)
+		if len(parts) != 3 {
+			continue
+		}
+
+		cgroupPath := parts[2]
+
+		// Check for direct container ID match in cgroup path
+		if container.ID != "" && strings.Contains(cgroupPath, container.ID) {
+			return true
+		}
+
+		// Check for container cgroup path match (for more complex hierarchies)
+		if container.CgroupPath != "" && strings.Contains(cgroupPath, container.CgroupPath) {
+			return true
+		}
+
+		// Check for runtime-specific patterns
+		switch container.Runtime {
+		case runtimev1.ContainerRuntime_CONTAINER_RUNTIME_DOCKER:
+			if strings.Contains(cgroupPath, "docker") && strings.Contains(cgroupPath, container.ID) {
+				return true
+			}
+		case runtimev1.ContainerRuntime_CONTAINER_RUNTIME_CONTAINERD, runtimev1.ContainerRuntime_CONTAINER_RUNTIME_CRI_CONTAINERD:
+			if strings.Contains(cgroupPath, "containerd") && strings.Contains(cgroupPath, container.ID) {
+				return true
+			}
+		case runtimev1.ContainerRuntime_CONTAINER_RUNTIME_CRI_O:
+			if strings.Contains(cgroupPath, "crio") && strings.Contains(cgroupPath, container.ID) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// createContainerCPURelationships creates relationships between container and CPU cores based on cpuset
+func (b *Builder) createContainerCPURelationships(containerRef *resourcev1.ResourceRef, cpusetCpus string) error {
+	// Parse CPU list using our shared utility
+	cpuList, err := performance.ParseCPUList(cpusetCpus)
+	if err != nil {
+		return fmt.Errorf("failed to parse cpuset_cpus %s: %w", cpusetCpus, err)
+	}
+
+	// Create relationships to each CPU core
+	for _, cpuID := range cpuList {
+		cpuRef := &resourcev1.ResourceRef{
+			TypeUrl: "hardware.antimetal.com/v1/CPU",
+			Name:    fmt.Sprintf("cpu-%d", cpuID),
+		}
+
+		if err := b.createContainerHardwareRelationship(containerRef, cpuRef); err != nil {
+			b.logger.Error(err, "Failed to create container-CPU relationship",
+				"container", containerRef.Name, "cpu", cpuID)
+		}
+	}
+
+	return nil
+}
+
+// createContainerNUMARelationships creates relationships between container and NUMA nodes based on cpuset
+func (b *Builder) createContainerNUMARelationships(containerRef *resourcev1.ResourceRef, cpusetMems string) error {
+	// Parse NUMA node list using our shared utility
+	numaList, err := performance.ParseCPUList(cpusetMems)
+	if err != nil {
+		return fmt.Errorf("failed to parse cpuset_mems %s: %w", cpusetMems, err)
+	}
+
+	// Create relationships to each NUMA node
+	for _, numaID := range numaList {
+		numaRef := &resourcev1.ResourceRef{
+			TypeUrl: "hardware.antimetal.com/v1/NUMANode",
```

Comment:
type url should be a valid proto type url (doesn't contain slashes)

Reply:
Ok, i'm now going to just use constants generated by reflection